src/Module.ts

```diff
@@ -463,8 +464,6 @@ export default class Module {
 		if (this.transitiveReexports) {
 			return this.transitiveReexports;
 		}
-		// to avoid infinite recursion when using circular `export * from X`
-		this.transitiveReexports = [];
```

Comment:
I would very much expect this one to be needed, maybe a test is missing. After all, there is a recursion in line 475 which needs to be broken if there is a cycle. Will need to look into this.

Reply:
I pushed a failing test for this one

Reply:
awesome! thank you! I'll revert.